File: src/main/java/com/lalkalol/mujahid/commands/impl/PauseCommand.java
package com.lalkalol.mujahid.commands.impl;

import com.lalkalol.mujahid.commands.Command;
import com.lalkalol.mujahid.commands.CommandContext;
import net.dv8tion.jda.api.interactions.commands.build.Commands;
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData;

public class PauseCommand implements Command {
    @Override
    public String name() {
        return "pause";
    }

    @Override
    public SlashCommandData data() {
        return Commands.slash("pause", "Pause the current track.");
    }

    @Override
    public void execute(CommandContext ctx) {
        if (!ctx.requireActivePlayer()) {
            return;
        }
        var player = ctx.getMusicManager().getPlayer();
        if (player != null && player.getPaused()) {
            ctx.replyEphemeral("Already paused.");
            return;
        }
        var link = ctx.getMusicManager().getCachedLink();
        if (link != null) {
            link.createOrUpdatePlayer().setPaused(true).subscribe();
        }
        ctx.reply("Paused.");
    }
}

File: src/main/kotlin/com/lalkalol/mujahid/commands/impl/PauseCommand.kt
package com.lalkalol.mujahid.commands.impl

import com.lalkalol.mujahid.commands.Command
import com.lalkalol.mujahid.commands.CommandContext
import net.dv8tion.jda.api.interactions.commands.build.Commands
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData

class PauseCommand : Command {
    override fun name(): String = "pause"

    override fun data(): SlashCommandData =
        Commands.slash("pause", "Pause the current track.")

    override suspend fun execute(ctx: CommandContext) {
        if (!ctx.requireActivePlayer()) {
            return
        }
        val player = ctx.getMusicManager().getPlayer()
        if (player != null && player.paused) {
            ctx.replyEphemeral("Already paused.")
            return
        }
        ctx.getMusicManager().getCachedLink()?.createOrUpdatePlayer()?.setPaused(true)?.subscribe()
        ctx.reply("Paused.")
    }
}
